spec/unit/data_providers/hiera_data_provider_spec.rb

```diff
@@ -189,7 +187,179 @@ def compile(environment, code = nil)
+        expect(lookup_invocation.explainer.to_s).to eq(<<EOS)
+Using merge options from 'lookup_options' hash
+Merge strategy deep
+  Data Binding "hiera"
+    No such key: "one::loptsm_test::hash"
+  Data Provider "Hiera Data Provider, version 4"
+    ConfigurationPath "#{environmentpath}/hiera_misc/hiera.yaml"
+    Data Provider "common"
+      Path "#{environmentpath}/hiera_misc/data/common.yaml"
+        Original path: common
+        Found key: "one::loptsm_test::hash" value: {
+          "a" => "A",
+          "b" => "B",
+          "m" => {
+            "ma" => "MA",
+            "mb" => "MB"
+          }
+        }
+  Module "one" using Data Provider "Hiera Data Provider, version 4"
+    ConfigurationPath "#{environmentpath}/hiera_misc/modules/one/hiera.yaml"
+    Data Provider "common"
+      Path "#{environmentpath}/hiera_misc/modules/one/data/common.yaml"
+        Original path: common
+        Found key: "one::loptsm_test::hash" value: {
+          "a" => "A",
+          "c" => "C",
+          "m" => {
+            "ma" => "MA",
+            "mc" => "MC",
+            "mb" => "MB"
+          }
+        }
+  Merged result: {
+    "a" => "A",
+    "c" => "C",
+    "m" => {
+      "ma" => "MA",
+      "mc" => "MC",
+      "mb" => "MB"
+    },
+    "b" => "B"
+  }
+EOS
+      end
+    end
+
+    it 'will report lookup_options details in combination with details of found value' do
+      compile('hiera_misc', '$target_scope = "with scope"') do |compiler|
+        lookup_invocation = Puppet::Pops::Lookup::Invocation.new(compiler.topscope, {}, {}, Puppet::Pops::Lookup::Explainer.new(true))
+        value = Puppet::Pops::Lookup.lookup('one::loptsm_test::hash', nil, nil, nil, nil, lookup_invocation)
+        expect(lookup_invocation.explainer.to_s).to eq(<<EOS)
+Searching for 'lookup_options'
+  Merge strategy hash
+    Data Binding "hiera"
+      No such key: "lookup_options"
+    Data Provider "Hiera Data Provider, version 4"
+      ConfigurationPath "#{environmentpath}/hiera_misc/hiera.yaml"
+      Data Provider "common"
+        Path "#{environmentpath}/hiera_misc/data/common.yaml"
+          Original path: common
+          Found key: "lookup_options" value: {
+            "one::lopts_test::hash" => {
+              "merge" => "deep"
+            }
+          }
+    Module "one" using Data Provider "Hiera Data Provider, version 4"
+      ConfigurationPath "#{environmentpath}/hiera_misc/modules/one/hiera.yaml"
+      Data Provider "common"
+        Path "#{environmentpath}/hiera_misc/modules/one/data/common.yaml"
+          Original path: common
+          Found key: "lookup_options" value: {
+            "one::loptsm_test::hash" => {
+              "merge" => "deep"
+            }
+          }
+    Merged result: {
+      "one::loptsm_test::hash" => {
+        "merge" => "deep"
+      },
+      "one::lopts_test::hash" => {
+        "merge" => "deep"
+      }
+    }
+Merge strategy deep
```

Comment:
a "Searching for 'one::loptsm_test::hash'" here would be really helpful in understanding that it is now reporting for the actual lookup.